tfjs-backend-webgpu/src/binary_op_util.ts

```diff
@@ -197,7 +186,9 @@ const POW_VEC4 = `
     resultTemp.a = 1.0;
   }
   let isNaN = (a < vec4<f32>(0.0)) & (floor(b) < b);
-  return select(resultTemp, vec4<f32>(uniforms.NAN), isNaN);
+  let valueForNaN = uniforms.NAN;
+  ${CHECK_NAN_SNIPPET_VEC4}
+  return resultTemp;
```

Comment:
The original code only checks `isNan`. Here you also add `isnanVec4(a) | isnanVec4(b)`. Is it expected?

Reply:
It's almost certain `binOp(NaN, b)` or `binOp(a, NaN)` will return `valueForNaN`. I guess previously this case was just quietly covered by the `pow()` builtin.